<h2 style="color:darkblue;">🏢 Análise de Funcionários - Empresa XYZ</h2>

<p style="color:gray; font-size:15px;">
Este notebook realiza uma análise exploratória dos dados de funcionários da Empresa XYZ, com o objetivo de entender a distribuição de cargos, salários, departamentos e tempo de casa.
</p>

<h4 style="color:darkgreen;">📁 Arquivo utilizado:</h4>
<span style="background-color:black; color:white; padding:6px 10px; border-radius:4px; font-family:monospace;">
funcionarios.csv
</span>

<h4 style="color:darkgreen;">📊 Principais colunas do dataset:</h4>
<ul>
  <li><strong>nome</strong>: Nome do funcionário</li>
  <li><strong>cargo</strong>: Função ou título</li>
  <li><strong>departamento</strong>: Área de atuação</li>
  <li><strong>salario</strong>: Remuneração mensal</li>
  <li><strong>data_admissao</strong>: Data de entrada na empresa</li>
</ul>

<h4 style="color:darkgreen;">🎯 Objetivos da análise:</h4>
<ul>
  <li>Identificar distribuição de salários por departamento</li>
  <li>Detectar cargos com maior rotatividade</li>
  <li>Visualizar tempo médio de permanência</li>
  <li>Gerar gráficos para insights gerenciais</li>
</ul>

<p style="background-color:#f0f8ff; color:#333; padding:10px; border-left:5px solid #007acc;">
💡 <strong>Dica:</strong> Use <code>df.describe()</code>, <code>df.groupby()</code> e <code>df.value_counts()</code> para explorar os dados.
</p>

## Importação da Biblioteca

In [1]:
import pandas as pd

## Lendo tabelas

In [2]:
df_funcionarios = pd.read_csv('funcionarios.csv')

In [3]:
df_funcionarios

ID                       Nome  Idade Data de Registro  Salário  Ativo  \
0    1   Sra. Isabela Vasconcelos     44       2022-02-11  4435.13   True   
1    2            Heitor Teixeira     40       2022-01-17  7834.97  False   
2    3         Bianca Vasconcelos     44       2022-10-28  3094.55  False   
3    4              Alice Machado     40       2023-10-10  6287.20   True   
4    5          Antônio Fernandes     54       2023-08-09  7138.43   True   
5    6      Henry Gabriel da Mata     42       2024-01-26  4941.62  False   
6    7  Sra. Maria Luiza Oliveira     58       2024-08-24  7111.25   True   
7    8               Isis Marques     24       2023-02-05  8438.66   True   
8    9          Sr. Vitor da Rosa     61       2024-03-25  2363.85  False   
9   10               Théo Cardoso     54       2022-09-05  8029.65   True   
10  11          Marcelo Rodrigues     57       2024-02-06  8109.35   True   
11  12         Maria Clara Santos     39       2023-10-21  2601.68   True   
12  13                 Sofia Rios     45       2023-05-19  4990.01  False   
13  14          Benicio das Neves     22       2022-04-15  2541.54   True   
14  15              Levi Silveira     42       2024-02-10  9261.98   True   
15  16     Maria Fernanda Rezende     29       2023-11-18  7436.21  False   
16  17         Thomas Casa Grande     33       2024-03-29  8205.90  False   
17  18        Davi Miguel Ribeiro     48       2022-07-10  2625.01  False   
18  19     Dr. Mathias Montenegro     20       2023-08-17  9566.83  False   
19  20           Samuel Gonçalves     21       2023-10-03  3884.99   True   

    ID Departamento  
0                 4  
1                 2  
2                 2  
3                 2  
4                 4  
5                 4  
6                 5  
7                 1  
8                 1  
9                 2  
10                1  
11                4  
12                1  
13                5  
14                5  
15                3  
16                2  
17                5  
18                4  
19                3

## Manipulação Básica de dados

## Manipulação do 5 primeiros registros

In [4]:
df_funcionarios.head(5)

ID                      Nome  Idade Data de Registro  Salário  Ativo  \
0   1  Sra. Isabela Vasconcelos     44       2022-02-11  4435.13   True   
1   2           Heitor Teixeira     40       2022-01-17  7834.97  False   
2   3        Bianca Vasconcelos     44       2022-10-28  3094.55  False   
3   4             Alice Machado     40       2023-10-10  6287.20   True   
4   5         Antônio Fernandes     54       2023-08-09  7138.43   True   

   ID Departamento  
0                4  
1                2  
2                2  
3                2  
4                4

## Manipulação dos 5 últimos registros

In [6]:
df_funcionarios.tail()

ID                    Nome  Idade Data de Registro  Salário  Ativo  \
15  16  Maria Fernanda Rezende     29       2023-11-18  7436.21  False   
16  17      Thomas Casa Grande     33       2024-03-29  8205.90  False   
17  18     Davi Miguel Ribeiro     48       2022-07-10  2625.01  False   
18  19  Dr. Mathias Montenegro     20       2023-08-17  9566.83  False   
19  20        Samuel Gonçalves     21       2023-10-03  3884.99   True   

    ID Departamento  
15                3  
16                2  
17                5  
18                4  
19                3

## Estatísticas dos funcionários

In [9]:
df_funcionarios.describe()

ID      Idade      Salário  ID Departamento
count  20.00000  20.000000    20.000000        20.000000
mean   10.50000  40.850000  5944.940500         3.000000
std     5.91608  12.716649  2474.638644         1.486784
min     1.00000  20.000000  2363.850000         1.000000
25%     5.75000  32.000000  3687.380000         2.000000
50%    10.50000  42.000000  6699.225000         3.000000
75%    15.25000  49.500000  8049.575000         4.000000
max    20.00000  61.000000  9566.830000         5.000000

## Colunas do dataset

In [10]:
df_funcionarios.columns

Index(['ID', 'Nome', 'Idade', 'Data de Registro', 'Salário', 'Ativo',
       'ID Departamento'],
      dtype='object')

## Informações sobre linhas e colunas do dataset

In [11]:
df_funcionarios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   ID                20 non-null     int64  
 1   Nome              20 non-null     object 
 2   Idade             20 non-null     int64  
 3   Data de Registro  20 non-null     object 
 4   Salário           20 non-null     float64
 5   Ativo             20 non-null     bool   
 6   ID Departamento   20 non-null     int64  
dtypes: bool(1), float64(1), int64(3), object(2)
memory usage: 1.1+ KB


## Selecionar o nome e a idade dos funcionários

In [12]:
df_funcionarios[['Nome', 'Idade']]

Nome  Idade
0    Sra. Isabela Vasconcelos     44
1             Heitor Teixeira     40
2          Bianca Vasconcelos     44
3               Alice Machado     40
4           Antônio Fernandes     54
5       Henry Gabriel da Mata     42
6   Sra. Maria Luiza Oliveira     58
7                Isis Marques     24
8           Sr. Vitor da Rosa     61
9                Théo Cardoso     54
10          Marcelo Rodrigues     57
11         Maria Clara Santos     39
12                 Sofia Rios     45
13          Benicio das Neves     22
14              Levi Silveira     42
15     Maria Fernanda Rezende     29
16         Thomas Casa Grande     33
17        Davi Miguel Ribeiro     48
18     Dr. Mathias Montenegro     20
19           Samuel Gonçalves     21

## Selecionar a idade dos funcionários maior que 30 anos

In [14]:
df_funcionarios[df_funcionarios['Idade'] > 30]

ID                       Nome  Idade Data de Registro  Salário  Ativo  \
0    1   Sra. Isabela Vasconcelos     44       2022-02-11  4435.13   True   
1    2            Heitor Teixeira     40       2022-01-17  7834.97  False   
2    3         Bianca Vasconcelos     44       2022-10-28  3094.55  False   
3    4              Alice Machado     40       2023-10-10  6287.20   True   
4    5          Antônio Fernandes     54       2023-08-09  7138.43   True   
5    6      Henry Gabriel da Mata     42       2024-01-26  4941.62  False   
6    7  Sra. Maria Luiza Oliveira     58       2024-08-24  7111.25   True   
8    9          Sr. Vitor da Rosa     61       2024-03-25  2363.85  False   
9   10               Théo Cardoso     54       2022-09-05  8029.65   True   
10  11          Marcelo Rodrigues     57       2024-02-06  8109.35   True   
11  12         Maria Clara Santos     39       2023-10-21  2601.68   True   
12  13                 Sofia Rios     45       2023-05-19  4990.01  False   
14  15              Levi Silveira     42       2024-02-10  9261.98   True   
16  17         Thomas Casa Grande     33       2024-03-29  8205.90  False   
17  18        Davi Miguel Ribeiro     48       2022-07-10  2625.01  False   

    ID Departamento  
0                 4  
1                 2  
2                 2  
3                 2  
4                 4  
5                 4  
6                 5  
8                 1  
9                 2  
10                1  
11                4  
12                1  
14                5  
16                2  
17                5

## Selecionar os salários dos funcionários maior que 5k

In [16]:
df_funcionarios[df_funcionarios['Salário'] > 5000]

ID                       Nome  Idade Data de Registro  Salário  Ativo  \
1    2            Heitor Teixeira     40       2022-01-17  7834.97  False   
3    4              Alice Machado     40       2023-10-10  6287.20   True   
4    5          Antônio Fernandes     54       2023-08-09  7138.43   True   
6    7  Sra. Maria Luiza Oliveira     58       2024-08-24  7111.25   True   
7    8               Isis Marques     24       2023-02-05  8438.66   True   
9   10               Théo Cardoso     54       2022-09-05  8029.65   True   
10  11          Marcelo Rodrigues     57       2024-02-06  8109.35   True   
14  15              Levi Silveira     42       2024-02-10  9261.98   True   
15  16     Maria Fernanda Rezende     29       2023-11-18  7436.21  False   
16  17         Thomas Casa Grande     33       2024-03-29  8205.90  False   
18  19     Dr. Mathias Montenegro     20       2023-08-17  9566.83  False   

    ID Departamento  
1                 2  
3                 2  
4                 4  
6                 5  
7                 1  
9                 2  
10                1  
14                5  
15                3  
16                2  
18                4

## Combinando filtros de idade e salário dos funcionários

In [18]:
filtro_maior_que_30 = df_funcionarios['Idade'] > 30
filtro_mais_de_5k = df_funcionarios['Salário'] > 5000

df_funcionarios[filtro_maior_que_30 & filtro_mais_de_5k]

ID                       Nome  Idade Data de Registro  Salário  Ativo  \
1    2            Heitor Teixeira     40       2022-01-17  7834.97  False   
3    4              Alice Machado     40       2023-10-10  6287.20   True   
4    5          Antônio Fernandes     54       2023-08-09  7138.43   True   
6    7  Sra. Maria Luiza Oliveira     58       2024-08-24  7111.25   True   
9   10               Théo Cardoso     54       2022-09-05  8029.65   True   
10  11          Marcelo Rodrigues     57       2024-02-06  8109.35   True   
14  15              Levi Silveira     42       2024-02-10  9261.98   True   
16  17         Thomas Casa Grande     33       2024-03-29  8205.90  False   

    ID Departamento  
1                 2  
3                 2  
4                 4  
6                 5  
9                 2  
10                1  
14                5  
16                2

## Criando uma nova coluna de total da soma de salários anuais dos funcionários

In [28]:
df_funcionarios['Salário_Anual'] = df_funcionarios['Salário'] * 12
df_funcionarios

ID                       Nome  Idade Data de Registro  Salário  Ativo  \
0    1   Sra. Isabela Vasconcelos     44       2022-02-11  4435.13   True   
1    2            Heitor Teixeira     40       2022-01-17  7834.97  False   
2    3         Bianca Vasconcelos     44       2022-10-28  3094.55  False   
3    4              Alice Machado     40       2023-10-10  6287.20   True   
4    5          Antônio Fernandes     54       2023-08-09  7138.43   True   
5    6      Henry Gabriel da Mata     42       2024-01-26  4941.62  False   
6    7  Sra. Maria Luiza Oliveira     58       2024-08-24  7111.25   True   
7    8               Isis Marques     24       2023-02-05  8438.66   True   
8    9          Sr. Vitor da Rosa     61       2024-03-25  2363.85  False   
9   10               Théo Cardoso     54       2022-09-05  8029.65   True   
10  11          Marcelo Rodrigues     57       2024-02-06  8109.35   True   
11  12         Maria Clara Santos     39       2023-10-21  2601.68   True   
12  13                 Sofia Rios     45       2023-05-19  4990.01  False   
13  14          Benicio das Neves     22       2022-04-15  2541.54   True   
14  15              Levi Silveira     42       2024-02-10  9261.98   True   
15  16     Maria Fernanda Rezende     29       2023-11-18  7436.21  False   
16  17         Thomas Casa Grande     33       2024-03-29  8205.90  False   
17  18        Davi Miguel Ribeiro     48       2022-07-10  2625.01  False   
18  19     Dr. Mathias Montenegro     20       2023-08-17  9566.83  False   
19  20           Samuel Gonçalves     21       2023-10-03  3884.99   True   

    ID Departamento  Salário_Anual  
0                 4       53221.56  
1                 2       94019.64  
2                 2       37134.60  
3                 2       75446.40  
4                 4       85661.16  
5                 4       59299.44  
6                 5       85335.00  
7                 1      101263.92  
8                 1       28366.20  
9                 2       96355.80  
10                1       97312.20  
11                4       31220.16  
12                1       59880.12  
13                5       30498.48  
14                5      111143.76  
15                3       89234.52  
16                2       98470.80  
17                5       31500.12  
18                4      114801.96  
19                3       46619.88

## Remoção de coluna

In [29]:
df_funcionarios = df_funcionarios.drop('Salário_Anual' , axis=1)
df_funcionarios

ID                       Nome  Idade Data de Registro  Salário  Ativo  \
0    1   Sra. Isabela Vasconcelos     44       2022-02-11  4435.13   True   
1    2            Heitor Teixeira     40       2022-01-17  7834.97  False   
2    3         Bianca Vasconcelos     44       2022-10-28  3094.55  False   
3    4              Alice Machado     40       2023-10-10  6287.20   True   
4    5          Antônio Fernandes     54       2023-08-09  7138.43   True   
5    6      Henry Gabriel da Mata     42       2024-01-26  4941.62  False   
6    7  Sra. Maria Luiza Oliveira     58       2024-08-24  7111.25   True   
7    8               Isis Marques     24       2023-02-05  8438.66   True   
8    9          Sr. Vitor da Rosa     61       2024-03-25  2363.85  False   
9   10               Théo Cardoso     54       2022-09-05  8029.65   True   
10  11          Marcelo Rodrigues     57       2024-02-06  8109.35   True   
11  12         Maria Clara Santos     39       2023-10-21  2601.68   True   
12  13                 Sofia Rios     45       2023-05-19  4990.01  False   
13  14          Benicio das Neves     22       2022-04-15  2541.54   True   
14  15              Levi Silveira     42       2024-02-10  9261.98   True   
15  16     Maria Fernanda Rezende     29       2023-11-18  7436.21  False   
16  17         Thomas Casa Grande     33       2024-03-29  8205.90  False   
17  18        Davi Miguel Ribeiro     48       2022-07-10  2625.01  False   
18  19     Dr. Mathias Montenegro     20       2023-08-17  9566.83  False   
19  20           Samuel Gonçalves     21       2023-10-03  3884.99   True   

    ID Departamento  
0                 4  
1                 2  
2                 2  
3                 2  
4                 4  
5                 4  
6                 5  
7                 1  
8                 1  
9                 2  
10                1  
11                4  
12                1  
13                5  
14                5  
15                3  
16                2  
17                5  
18                4  
19                3

In [30]:
print(df_funcionarios.columns)

Index(['ID', 'Nome', 'Idade', 'Data de Registro', 'Salário', 'Ativo',
       'ID Departamento'],
      dtype='object')


## Soma total da coluna salário

In [31]:
df_funcionarios['Salário'].sum()

118898.80999999998

## Soma média da coluna salário

In [32]:
df_funcionarios['Salário'].mean()

5944.940499999999

## Soma do maior salário entre os funcionários

In [33]:
df_funcionarios['Salário'].max()

9566.83

## Soma do menor salário entre os funcionários

In [34]:
df_funcionarios['Salário'].min()

2363.85

## Código para analisar funcionários ativos

In [44]:
df_funcionarios['Ativo'].value_counts()

Ativo
True     11
False     9
Name: count, dtype: int64

## Contar funcionários através do id por departamento

In [46]:
df_funcionarios['ID Departamento'].value_counts()

ID Departamento
4    5
2    5
5    4
1    4
3    2
Name: count, dtype: int64

## Cálculo de Métricas do id departamento

In [48]:
df_funcionarios.groupby('ID Departamento')

In [49]:
df_funcionarios.groupby('ID Departamento')['Salário'].mean()

ID Departamento
1    5975.4675
2    6690.4540
3    5660.6000
4    5736.7380
5    5384.9450
Name: Salário, dtype: float64

In [50]:
df_funcionarios.groupby('ID Departamento')['Salário'].max()

ID Departamento
1    8438.66
2    8205.90
3    7436.21
4    9566.83
5    9261.98
Name: Salário, dtype: float64

In [51]:
df_funcionarios.groupby('ID Departamento')['Salário'].min()

ID Departamento
1    2363.85
2    3094.55
3    3884.99
4    2601.68
5    2541.54
Name: Salário, dtype: float64

## Odenar salários do menor para o maior

In [52]:
df_funcionarios.sort_values(by='Salário')

ID                       Nome  Idade Data de Registro  Salário  Ativo  \
8    9          Sr. Vitor da Rosa     61       2024-03-25  2363.85  False   
13  14          Benicio das Neves     22       2022-04-15  2541.54   True   
11  12         Maria Clara Santos     39       2023-10-21  2601.68   True   
17  18        Davi Miguel Ribeiro     48       2022-07-10  2625.01  False   
2    3         Bianca Vasconcelos     44       2022-10-28  3094.55  False   
19  20           Samuel Gonçalves     21       2023-10-03  3884.99   True   
0    1   Sra. Isabela Vasconcelos     44       2022-02-11  4435.13   True   
5    6      Henry Gabriel da Mata     42       2024-01-26  4941.62  False   
12  13                 Sofia Rios     45       2023-05-19  4990.01  False   
3    4              Alice Machado     40       2023-10-10  6287.20   True   
6    7  Sra. Maria Luiza Oliveira     58       2024-08-24  7111.25   True   
4    5          Antônio Fernandes     54       2023-08-09  7138.43   True   
15  16     Maria Fernanda Rezende     29       2023-11-18  7436.21  False   
1    2            Heitor Teixeira     40       2022-01-17  7834.97  False   
9   10               Théo Cardoso     54       2022-09-05  8029.65   True   
10  11          Marcelo Rodrigues     57       2024-02-06  8109.35   True   
16  17         Thomas Casa Grande     33       2024-03-29  8205.90  False   
7    8               Isis Marques     24       2023-02-05  8438.66   True   
14  15              Levi Silveira     42       2024-02-10  9261.98   True   
18  19     Dr. Mathias Montenegro     20       2023-08-17  9566.83  False   

    ID Departamento  
8                 1  
13                5  
11                4  
17                5  
2                 2  
19                3  
0                 4  
5                 4  
12                1  
3                 2  
6                 5  
4                 4  
15                3  
1                 2  
9                 2  
10                1  
16                2  
7                 1  
14                5  
18                4

## Ordenar salário do maior para o menor

In [54]:
df_funcionarios.sort_values(by='Salário', ascending=False)

ID                       Nome  Idade Data de Registro  Salário  Ativo  \
18  19     Dr. Mathias Montenegro     20       2023-08-17  9566.83  False   
14  15              Levi Silveira     42       2024-02-10  9261.98   True   
7    8               Isis Marques     24       2023-02-05  8438.66   True   
16  17         Thomas Casa Grande     33       2024-03-29  8205.90  False   
10  11          Marcelo Rodrigues     57       2024-02-06  8109.35   True   
9   10               Théo Cardoso     54       2022-09-05  8029.65   True   
1    2            Heitor Teixeira     40       2022-01-17  7834.97  False   
15  16     Maria Fernanda Rezende     29       2023-11-18  7436.21  False   
4    5          Antônio Fernandes     54       2023-08-09  7138.43   True   
6    7  Sra. Maria Luiza Oliveira     58       2024-08-24  7111.25   True   
3    4              Alice Machado     40       2023-10-10  6287.20   True   
12  13                 Sofia Rios     45       2023-05-19  4990.01  False   
5    6      Henry Gabriel da Mata     42       2024-01-26  4941.62  False   
0    1   Sra. Isabela Vasconcelos     44       2022-02-11  4435.13   True   
19  20           Samuel Gonçalves     21       2023-10-03  3884.99   True   
2    3         Bianca Vasconcelos     44       2022-10-28  3094.55  False   
17  18        Davi Miguel Ribeiro     48       2022-07-10  2625.01  False   
11  12         Maria Clara Santos     39       2023-10-21  2601.68   True   
13  14          Benicio das Neves     22       2022-04-15  2541.54   True   
8    9          Sr. Vitor da Rosa     61       2024-03-25  2363.85  False   

    ID Departamento  
18                4  
14                5  
7                 1  
16                2  
10                1  
9                 2  
1                 2  
15                3  
4                 4  
6                 5  
3                 2  
12                1  
5                 4  
0                 4  
19                3  
2                 2  
17                5  
11                4  
13                5  
8                 1

## Limpeza de dados

In [35]:
df_nulos = pd.read_csv('funcionarios_nulos.csv')
df_nulos

ID                     Nome  Idade Data de Registro  Salário  Ativo  \
0    1                      NaN   35.0       2024-01-02  5469.47   True   
1    2          Nicolas Cardoso   45.0       2022-05-25  3824.20   True   
2    3    Lucas Gabriel Sampaio   60.0       2023-01-05  5599.99   True   
3    4  Luiz Fernando Fernandes    NaN       2023-10-09  6708.07  False   
4    5    Gael Henrique Costela   51.0       2021-11-07  9497.52    NaN   
5    6         Stephany Pimenta   32.0       2024-07-21  7123.00   True   
6    7             Bruna Vargas   25.0       2022-03-29  5036.81    NaN   
7    8                      NaN   60.0       2024-03-03  8219.79   True   
8    9                      NaN   47.0       2022-09-12  4790.53    NaN   
9   10                      NaN    NaN       2022-06-26  8914.27    NaN   
10  11                      NaN    NaN       2023-02-04  4475.88    NaN   
11  12           Manuela Guerra    NaN       2022-03-31  9883.09  False   
12  13        Dr. Apollo Moraes    NaN       2023-01-16  5392.79   True   
13  14        Antônio Gonçalves   44.0              NaN  5511.17   True   
14  15               Luara Lima   48.0              NaN  5426.62   True   
15  16                      NaN   26.0              NaN  6519.70  False   
16  17                      NaN   41.0       2024-07-11  7815.03  False   
17  18           Isabela Campos   28.0       2022-06-21  7824.41  False   
18  19             Danilo Rocha   32.0       2024-08-07  7196.97  False   
19  20             Sara Fonseca   50.0       2023-09-05  5884.55   True   

    ID Departamento  
0               3.0  
1               4.0  
2               4.0  
3               NaN  
4               NaN  
5               NaN  
6               NaN  
7               5.0  
8               4.0  
9               4.0  
10              NaN  
11              NaN  
12              NaN  
13              NaN  
14              3.0  
15              NaN  
16              5.0  
17              4.0  
18              4.0  
19              2.0

## Código para remover todos os NaN

In [37]:
df_nulos.dropna()

ID                   Nome  Idade Data de Registro  Salário  Ativo  \
1    2        Nicolas Cardoso   45.0       2022-05-25  3824.20   True   
2    3  Lucas Gabriel Sampaio   60.0       2023-01-05  5599.99   True   
17  18         Isabela Campos   28.0       2022-06-21  7824.41  False   
18  19           Danilo Rocha   32.0       2024-08-07  7196.97  False   
19  20           Sara Fonseca   50.0       2023-09-05  5884.55   True   

    ID Departamento  
1               4.0  
2               4.0  
17              4.0  
18              4.0  
19              2.0

## Removendo os NaN só da coluna idade

In [38]:
df_nulos.dropna(subset='Idade')

ID                   Nome  Idade Data de Registro  Salário  Ativo  \
0    1                    NaN   35.0       2024-01-02  5469.47   True   
1    2        Nicolas Cardoso   45.0       2022-05-25  3824.20   True   
2    3  Lucas Gabriel Sampaio   60.0       2023-01-05  5599.99   True   
4    5  Gael Henrique Costela   51.0       2021-11-07  9497.52    NaN   
5    6       Stephany Pimenta   32.0       2024-07-21  7123.00   True   
6    7           Bruna Vargas   25.0       2022-03-29  5036.81    NaN   
7    8                    NaN   60.0       2024-03-03  8219.79   True   
8    9                    NaN   47.0       2022-09-12  4790.53    NaN   
13  14      Antônio Gonçalves   44.0              NaN  5511.17   True   
14  15             Luara Lima   48.0              NaN  5426.62   True   
15  16                    NaN   26.0              NaN  6519.70  False   
16  17                    NaN   41.0       2024-07-11  7815.03  False   
17  18         Isabela Campos   28.0       2022-06-21  7824.41  False   
18  19           Danilo Rocha   32.0       2024-08-07  7196.97  False   
19  20           Sara Fonseca   50.0       2023-09-05  5884.55   True   

    ID Departamento  
0               3.0  
1               4.0  
2               4.0  
4               NaN  
5               NaN  
6               NaN  
7               5.0  
8               4.0  
13              NaN  
14              3.0  
15              NaN  
16              5.0  
17              4.0  
18              4.0  
19              2.0

## Preenchendo os valores da coluna idade

In [40]:
media_idade = df_nulos['Idade'].mean()
df_nulos['Idade'] = df_nulos['Idade'].fillna(media_idade)
df_nulos

ID                     Nome  Idade Data de Registro  Salário  Ativo  \
0    1                      NaN   35.0       2024-01-02  5469.47   True   
1    2          Nicolas Cardoso   45.0       2022-05-25  3824.20   True   
2    3    Lucas Gabriel Sampaio   60.0       2023-01-05  5599.99   True   
3    4  Luiz Fernando Fernandes   41.6       2023-10-09  6708.07  False   
4    5    Gael Henrique Costela   51.0       2021-11-07  9497.52    NaN   
5    6         Stephany Pimenta   32.0       2024-07-21  7123.00   True   
6    7             Bruna Vargas   25.0       2022-03-29  5036.81    NaN   
7    8                      NaN   60.0       2024-03-03  8219.79   True   
8    9                      NaN   47.0       2022-09-12  4790.53    NaN   
9   10                      NaN   41.6       2022-06-26  8914.27    NaN   
10  11                      NaN   41.6       2023-02-04  4475.88    NaN   
11  12           Manuela Guerra   41.6       2022-03-31  9883.09  False   
12  13        Dr. Apollo Moraes   41.6       2023-01-16  5392.79   True   
13  14        Antônio Gonçalves   44.0              NaN  5511.17   True   
14  15               Luara Lima   48.0              NaN  5426.62   True   
15  16                      NaN   26.0              NaN  6519.70  False   
16  17                      NaN   41.0       2024-07-11  7815.03  False   
17  18           Isabela Campos   28.0       2022-06-21  7824.41  False   
18  19             Danilo Rocha   32.0       2024-08-07  7196.97  False   
19  20             Sara Fonseca   50.0       2023-09-05  5884.55   True   

    ID Departamento  
0               3.0  
1               4.0  
2               4.0  
3               NaN  
4               NaN  
5               NaN  
6               NaN  
7               5.0  
8               4.0  
9               4.0  
10              NaN  
11              NaN  
12              NaN  
13              NaN  
14              3.0  
15              NaN  
16              5.0  
17              4.0  
18              4.0  
19              2.0

## Convertendo os números da idade 41.6 para números inteiros

In [42]:
df_nulos['Idade'] = df_nulos['Idade'].astype(int)
df_nulos

ID                     Nome  Idade Data de Registro  Salário  Ativo  \
0    1                      NaN     35       2024-01-02  5469.47   True   
1    2          Nicolas Cardoso     45       2022-05-25  3824.20   True   
2    3    Lucas Gabriel Sampaio     60       2023-01-05  5599.99   True   
3    4  Luiz Fernando Fernandes     41       2023-10-09  6708.07  False   
4    5    Gael Henrique Costela     51       2021-11-07  9497.52    NaN   
5    6         Stephany Pimenta     32       2024-07-21  7123.00   True   
6    7             Bruna Vargas     25       2022-03-29  5036.81    NaN   
7    8                      NaN     60       2024-03-03  8219.79   True   
8    9                      NaN     47       2022-09-12  4790.53    NaN   
9   10                      NaN     41       2022-06-26  8914.27    NaN   
10  11                      NaN     41       2023-02-04  4475.88    NaN   
11  12           Manuela Guerra     41       2022-03-31  9883.09  False   
12  13        Dr. Apollo Moraes     41       2023-01-16  5392.79   True   
13  14        Antônio Gonçalves     44              NaN  5511.17   True   
14  15               Luara Lima     48              NaN  5426.62   True   
15  16                      NaN     26              NaN  6519.70  False   
16  17                      NaN     41       2024-07-11  7815.03  False   
17  18           Isabela Campos     28       2022-06-21  7824.41  False   
18  19             Danilo Rocha     32       2024-08-07  7196.97  False   
19  20             Sara Fonseca     50       2023-09-05  5884.55   True   

    ID Departamento  
0               3.0  
1               4.0  
2               4.0  
3               NaN  
4               NaN  
5               NaN  
6               NaN  
7               5.0  
8               4.0  
9               4.0  
10              NaN  
11              NaN  
12              NaN  
13              NaN  
14              3.0  
15              NaN  
16              5.0  
17              4.0  
18              4.0  
19              2.0

## ID'S de departamentos

In [60]:
df_depto = pd.read_csv('departamentos.csv')
df_depto

ID Departamento Nome Departamento
0                1            Vendas
1                2                TI
2                3                RH
3                4        Financeiro
4                5         Marketing

## Código para juntar tabelas e datasets

In [61]:
df_final = pd.merge(df_funcionarios, df_depto)
df_final

ID                       Nome  Idade Data de Registro  Salário  Ativo  \
0    1   Sra. Isabela Vasconcelos     44       2022-02-11  4435.13   True   
1    5          Antônio Fernandes     54       2023-08-09  7138.43   True   
2    6      Henry Gabriel da Mata     42       2024-01-26  4941.62  False   
3   12         Maria Clara Santos     39       2023-10-21  2601.68   True   
4   19     Dr. Mathias Montenegro     20       2023-08-17  9566.83  False   
5    2            Heitor Teixeira     40       2022-01-17  7834.97  False   
6    3         Bianca Vasconcelos     44       2022-10-28  3094.55  False   
7    4              Alice Machado     40       2023-10-10  6287.20   True   
8   10               Théo Cardoso     54       2022-09-05  8029.65   True   
9   17         Thomas Casa Grande     33       2024-03-29  8205.90  False   
10   7  Sra. Maria Luiza Oliveira     58       2024-08-24  7111.25   True   
11  14          Benicio das Neves     22       2022-04-15  2541.54   True   
12  15              Levi Silveira     42       2024-02-10  9261.98   True   
13  18        Davi Miguel Ribeiro     48       2022-07-10  2625.01  False   
14   8               Isis Marques     24       2023-02-05  8438.66   True   
15   9          Sr. Vitor da Rosa     61       2024-03-25  2363.85  False   
16  11          Marcelo Rodrigues     57       2024-02-06  8109.35   True   
17  13                 Sofia Rios     45       2023-05-19  4990.01  False   
18  16     Maria Fernanda Rezende     29       2023-11-18  7436.21  False   
19  20           Samuel Gonçalves     21       2023-10-03  3884.99   True   

    ID Departamento Nome Departamento  
0                 4        Financeiro  
1                 4        Financeiro  
2                 4        Financeiro  
3                 4        Financeiro  
4                 4        Financeiro  
5                 2                TI  
6                 2                TI  
7                 2                TI  
8                 2                TI  
9                 2                TI  
10                5         Marketing  
11                5         Marketing  
12                5         Marketing  
13                5         Marketing  
14                1            Vendas  
15                1            Vendas  
16                1            Vendas  
17                1            Vendas  
18                3                RH  
19                3                RH

## Adicionar novos funcionários de outro dataset para o dataframe de um dataset

In [71]:
df_novos = pd.read_csv('novos_funcionarios.csv')
df_novos

ID                     Nome  Idade Data de Registro  Salário  Ativo  \
0    1        Maria Clara Cunha     27       2023-03-05  6747.76   True   
1    2      Luiz Gustavo Barros     18       2022-11-11  5990.81  False   
2    3   Maria Helena das Neves     19       2024-04-01  7588.55  False   
3    4             Ravy Camargo     36       2022-01-25  4341.11  False   
4    5      Cecilia Casa Grande     46       2022-10-30  5716.84   True   
5    6           Carolina Souza     62       2022-05-02  3305.85  False   
6    7            Mathias Souza     28       2024-05-02  9782.90   True   
7    8               Yago Gomes     28       2023-10-17  9615.18   True   
8    9           Natália Moraes     20       2022-10-03  2798.04   True   
9   10     João Miguel Caldeira     53       2024-03-12  8941.77   True   
10  11          Thales Monteiro     55       2022-05-20  7587.85   True   
11  12           Miguel Pacheco     61       2022-12-22  7690.91  False   
12  13         Léo da Conceição     40       2023-01-20  7552.60   True   
13  14          Aurora Carvalho     53       2022-03-27  2954.14   True   
14  15            Mathias Rocha     44       2024-01-18  9316.01   True   
15  16        Maria Liz Sampaio     34       2023-12-01  8334.63  False   
16  17  Arthur Gabriel Cassiano     42       2024-03-23  4275.55  False   
17  18              Joana Sousa     54       2023-05-20  5755.79   True   
18  19    Srta. Ana Lívia Silva     21       2023-11-19  7636.43  False   
19  20           Gustavo Castro     64       2022-02-04  3641.82   True   

    ID Departamento  
0                 4  
1                 3  
2                 5  
3                 4  
4                 4  
5                 4  
6                 4  
7                 1  
8                 5  
9                 1  
10                2  
11                2  
12                3  
13                1  
14                5  
15                2  
16                4  
17                5  
18                5  
19                2

In [72]:
df_todos = pd.concat([df_funcionarios, df_novos], ignore_index=True)
df_todos

ID                       Nome  Idade Data de Registro  Salário  Ativo  \
0    1   Sra. Isabela Vasconcelos     44       2022-02-11  4435.13   True   
1    2            Heitor Teixeira     40       2022-01-17  7834.97  False   
2    3         Bianca Vasconcelos     44       2022-10-28  3094.55  False   
3    4              Alice Machado     40       2023-10-10  6287.20   True   
4    5          Antônio Fernandes     54       2023-08-09  7138.43   True   
5    6      Henry Gabriel da Mata     42       2024-01-26  4941.62  False   
6    7  Sra. Maria Luiza Oliveira     58       2024-08-24  7111.25   True   
7    8               Isis Marques     24       2023-02-05  8438.66   True   
8    9          Sr. Vitor da Rosa     61       2024-03-25  2363.85  False   
9   10               Théo Cardoso     54       2022-09-05  8029.65   True   
10  11          Marcelo Rodrigues     57       2024-02-06  8109.35   True   
11  12         Maria Clara Santos     39       2023-10-21  2601.68   True   
12  13                 Sofia Rios     45       2023-05-19  4990.01  False   
13  14          Benicio das Neves     22       2022-04-15  2541.54   True   
14  15              Levi Silveira     42       2024-02-10  9261.98   True   
15  16     Maria Fernanda Rezende     29       2023-11-18  7436.21  False   
16  17         Thomas Casa Grande     33       2024-03-29  8205.90  False   
17  18        Davi Miguel Ribeiro     48       2022-07-10  2625.01  False   
18  19     Dr. Mathias Montenegro     20       2023-08-17  9566.83  False   
19  20           Samuel Gonçalves     21       2023-10-03  3884.99   True   
20   1          Maria Clara Cunha     27       2023-03-05  6747.76   True   
21   2        Luiz Gustavo Barros     18       2022-11-11  5990.81  False   
22   3     Maria Helena das Neves     19       2024-04-01  7588.55  False   
23   4               Ravy Camargo     36       2022-01-25  4341.11  False   
24   5        Cecilia Casa Grande     46       2022-10-30  5716.84   True   
25   6             Carolina Souza     62       2022-05-02  3305.85  False   
26   7              Mathias Souza     28       2024-05-02  9782.90   True   
27   8                 Yago Gomes     28       2023-10-17  9615.18   True   
28   9             Natália Moraes     20       2022-10-03  2798.04   True   
29  10       João Miguel Caldeira     53       2024-03-12  8941.77   True   
30  11            Thales Monteiro     55       2022-05-20  7587.85   True   
31  12             Miguel Pacheco     61       2022-12-22  7690.91  False   
32  13           Léo da Conceição     40       2023-01-20  7552.60   True   
33  14            Aurora Carvalho     53       2022-03-27  2954.14   True   
34  15              Mathias Rocha     44       2024-01-18  9316.01   True   
35  16          Maria Liz Sampaio     34       2023-12-01  8334.63  False   
36  17    Arthur Gabriel Cassiano     42       2024-03-23  4275.55  False   
37  18                Joana Sousa     54       2023-05-20  5755.79   True   
38  19      Srta. Ana Lívia Silva     21       2023-11-19  7636.43  False   
39  20             Gustavo Castro     64       2022-02-04  3641.82   True   

    ID Departamento  
0                 4  
1                 2  
2                 2  
3                 2  
4                 4  
5                 4  
6                 5  
7                 1  
8                 1  
9                 2  
10                1  
11                4  
12                1  
13                5  
14                5  
15                3  
16                2  
17                5  
18                4  
19                3  
20                4  
21                3  
22                5  
23                4  
24                4  
25                4  
26                4  
27                1  
28                5  
29                1  
30                2  
31                2  
32                3  
33                1  
34                5  
35                2  
36                4  
37        

## Salvando os dados dos dois datasets para outro dataset com todos o funcionários, os antigos e os novos

In [73]:
df_todos.to_csv('todos_funcionarios.csv')